stustapay/core/service/dbservice.py

```diff
@@ -42,30 +42,59 @@ async def wrapper(self, **kwargs):
     return wrapper
 
 
+async def _get_user_and_terminal(self, **kwargs) -> Tuple[Optional[User], Optional[Terminal]]:
```

Comment:
As stated in my other comment this method does not make much sense in my opinion as it combines two unrelated concepts. Terminals and users are not related (even though in both cases a user might be logged in).

The API used by terminals and the API used by the admin interface will always differ. Therefore every API using this decorator will internally again have some sort of conditional logic to do something depending on if the request came from a terminal or from the admin interface. We should stick with the two separate decorators and only move common code to internal helpers in the services if needed.